lutris/runners/runner.py

```diff
@@ -298,17 +298,26 @@ def get_runner_version(self, version=None):
             version (str): Optional version to lookup, will return this one if found
 
         Returns:
-            dict: Dict containing version, architecture and url for the runner
+            dict: Dict containing version, architecture and url for the runner, None
+            if the data can't be retrieved.
         """
         logger.info(
             "Getting runner information for %s%s",
             self.name,
             " (version: %s)" % version if version else "",
         )
-        request = Request("{}/api/runners/{}".format(settings.SITE_URL, self.name))
-        runner_info = request.get().json
+
+        try:
+            request = Request("{}/api/runners/{}".format(settings.SITE_URL, self.name))
+            runner_info = request.get().json
+
+            if not runner_info:
+                logger.error("Failed to get runner information")
```

Comment:
No point in repeating the if, just leave the logger.error call in the original if.

Reply:
Not quite sure what you mean here.

This code is a little bit odd because pylint thinks that 6 returns in a function is okay, but 7 is too many. I would have had two returns in this code but for that.

I didn't feel comfortable suppressing the pylint warning, or rewriting the whole function. So I did this.